src/components/portal/portal.tsx

```diff
@@ -212,24 +211,6 @@ export class Portal {
         this.parent.appendChild(this.container);
     }
 
-    private removeContainer() {
-        if (!this.container) {
-            return;
-        }
-
-        Array.from(this.container.children).forEach((element: HTMLElement) => {
-            const parent = this.parents.get(element);
-            if (!parent) {
-                return;
-            }
-
-            parent.appendChild(element);
-        });
```

Comment:
Most of the code changes in this commit is done after consultation with Chat GPT, after a long back and forth discussion about what I exactly wanted to happen. So I'm not sure about this code removal. When I asked why:

> The section you removed from the original removeContainer function was responsible for moving the elements inside the portal (children) back to their original parent in the DOM before the portal container itself was removed. Specifically, this part of the code:
> 
> **What it did:**
> - ` this.parents.get(element)`: It retrieves the original parent of the element, stored earlier in a `WeakMap`.
> - `parent.appendChild(element)`: It re-attaches each child back to its original parent in the DOM.
> **Consequences of removing it:**
> - If you are no longer moving elements out of their original position in the DOM (e.g., not manipulating their original DOM tree, just hiding and showing the portal), then this removal is safe.
> - If you are still moving elements out of the DOM (e.g., you are attaching elements into the portal container from other parts of the DOM), you might now be "losing" those elements when the container is removed, as they won’t be re-attached to their original parents.
> **When it's safe to remove:**
> - If your current code only manipulates the portal container (showing and hiding) without moving its children around in the DOM, then this removal is safe and aligns with your refactor.
> **When it's not safe:**
> - If you still need to move DOM elements into the portal container and expect them to return to their original place in the DOM when the portal is closed, this removal could lead to issues such as:
> - Elements disappearing permanently from the DOM when the container is removed.
> - Elements not being properly returned to their original positions.
> - If your current portal logic does not involve moving children from the original DOM tree, then you're fine. Otherwise, it may require revisiting this mechanism.


@jgroth this sounded similar or related to what you tried to do in this PR https://github.com/Lundalogik/lime-elements/pull/3182/ which is why I wanted us to wait with that one.

Reply:
I don't see why we should wait with the other one. It took 30 seconds to open the admin page due to 2000+ portals moving elements around in the DOM. It seems more important to fix that than adding animations. Don't get me wrong, animations are nice but kind of pointless if you cant even load the page in the first place :smile:

Reply:
ok. so merge that one, and I'll try to remake this based on the updated code.